src/main/java/org/skriptlang/skript/common/function/Signature.java

```diff
@@ -13,6 +19,32 @@
 @NonExtendable
 @Internal
 @Experimental
-public interface Signature<T> {
+public interface Signature<T> extends Documentable {
+
+	/**
+	 * @return The type of this parameter.
+	 */
+	Class<T> returnType();
+
+	/**
+	 * @return An unmodifiable view of all the parameters that this signature has.
+	 */
+	@Unmodifiable @NotNull SequencedMap<String, Parameter<?>> parameters();
```

Comment:
Why does this need to be a map? Don't parameters store their names?

Reply:
Parameters do store their names, but making this a map allows for easy lookup of parameters by name, something which happens often. This can of course be replaced with any sequenced non-map collection, but that would require having to loop through all parameters to get a parameter by name. This can be added as a helper method, but using a map here is probably faster (and easier).

As for it being sequenced, Signature needs to contain a method to get the order of the parameters, as function calls like `x(1, 2, 3)` should still be able to match arguments to parameters without having the arguments be named.